### PR TITLE
Zeroconf announce for http::Server

### PR DESCRIPTION
**Pull request by eile**

### CODE REVIEW

File: zeroeq/http/server.cpp [hunk 1]
```diff
@@ -176,6 +176,25 @@ class Server::Impl : public detail::Sender
...
+        const servus::Servus::Result& result =
+            _service.announce(uri.getPort(), getAddress());
+
+        if (!result)
+        {
+            ZEROEQTHROW(std::runtime_error("Zeroconf announce failed: " +
+                                           result.getString()));
+        }
+
         if (uri_.getHost() != uri.getHost() || uri_.getPort() == 0)
```
Comment (tribal-tec) [inline zeroeq/http/server.cpp]: this should be printed independently of the servus stuff

File: zeroeq/http/server.cpp [hunk 2]
```diff
@@ -78,6 +80,7 @@ std::string _convertEndpointName(const std::string& endpoint)
 const std::string JSON_TYPE = "application/json";
 const std::string REQUEST_REGISTRY = "registry";
 const std::string REQUEST_SCHEMA = "schema";
+const std::string HTTP_SERVER_SERVICE = "_http._tcp";
```
Comment (tribal-tec) [inline zeroeq/http/server.cpp]: _zeroeqhttp._tcp? otherwise it looks very generic.

Comment (tribal-tec) [inline zeroeq/http/server.cpp]: Also, add some documentation in the header of the class that this features exists.

Comment (eile) [inline zeroeq/http/server.cpp]: I on purpose used the generic service for HTTP servers.

Comment (tribal-tec) [inline zeroeq/http/server.cpp]: Ah ok. Then mention it for future reference :)

File: zeroeq/http/server.cpp (hunk 1 above)
Comment (eile) [inline zeroeq/http/server.cpp]: It is, unless Zeroconf announcement fails in which case the ctor throws anyways.

File: zeroeq/http/server.cpp (hunk 2 above)
Comment (eile) [inline zeroeq/http/server.cpp]: doc added

File: zeroeq/http/server.cpp (hunk 1 above)
Comment (eile) [inline zeroeq/http/server.cpp]: Correction, it isn't. Fixing.

Comment (eile) [inline zeroeq/http/server.cpp]: done

**Comment by tribal-tec** [comment]
+1 for me. @hernando also should a have look

File: zeroeq/http/server.cpp [hunk 3]
```diff
@@ -179,6 +179,25 @@ class Server::Impl : public detail::Sender
...
+        if (!_session.empty())
+            _service.set(KEY_SESSION, _session);
+
+        const servus::Servus::Result& result =
+            _service.announce(uri.getPort(), getAddress());
+
+        if (!result)
+        {
+            ZEROEQTHROW(std::runtime_error("Zeroconf announce failed: " +
+                                           result.getString()));
```
Comment (hernando) [inline zeroeq/http/server.cpp]: Do you really want to throw if the announcement fails?

Comment (eile) [inline zeroeq/http/server.cpp]: Since this is C&P from the Publisher, I guess so?

**Comment by eile** [comment]
> Explain a bit more the motivation in the commit message.

Done

**Pull request merged**